File: python/01_generate_index.py
"""
01_generate_index.py

Submits a build job to TensorTensor and prints the jobId.
"""

import os
import sys
import requests
from dotenv import load_dotenv

load_dotenv()

api_url     = os.getenv('TT_API_URL')
api_key     = os.getenv('TT_API_KEY')
vectors_url = os.getenv('TT_VECTORS_URL')
dimensions  = int(os.getenv('TT_DIMENSIONS', '0'))
email       = os.getenv('TT_EMAIL')


def assert_env():
    missing = [k for k, v in {
        'TT_API_URL': api_url,
        'TT_API_KEY': api_key,
        'TT_VECTORS_URL': vectors_url,
        'TT_DIMENSIONS': dimensions,
        'TT_EMAIL': email,
    }.items() if not v]
    if missing:
        print('Missing env vars:', ', '.join(missing), file=sys.stderr)
        sys.exit(1)


def main():
    assert_env()

    headers = {
        'Authorization': f'Basic {api_key}',
        'Content-Type': 'application/json',
    }
    payload = {
        'email': email,
        'vectorsUrl': vectors_url,
        'dimensions': dimensions,
    }

    res = requests.post(f'{api_url}/api/v1/build', json=payload, headers=headers)
    res.raise_for_status()
    data = res.json()['data']
    print('jobId :', data['jobId'])
    print('status:', data['status'])
    print('')
    print('Save the jobId. You will receive an email when the build completes,')
    print('or run 03_end_to_end.py for an automated submit-and-wait flow.')


if __name__ == '__main__':
    main()

File: typescript/01-generate-index.ts
/*
 * 01-generate-index.ts
 *
 * Submits a build job to TensorTensor and prints the jobId.
 */

import axios from 'axios';
import 'dotenv/config';

interface BuildResponse {
  data: {
    jobId: string;
    status: string;
  };
}

const apiUrl     = process.env.TT_API_URL;
const apiKey     = process.env.TT_API_KEY;
const vectorsUrl = process.env.TT_VECTORS_URL;
const dimensions = Number(process.env.TT_DIMENSIONS);
const email      = process.env.TT_EMAIL;

function assertEnv(): void {

  const missing: string[] = [];
  if (!apiUrl)     missing.push('TT_API_URL');
  if (!apiKey)     missing.push('TT_API_KEY');
  if (!vectorsUrl) missing.push('TT_VECTORS_URL');
  if (!dimensions) missing.push('TT_DIMENSIONS');
  if (!email)      missing.push('TT_EMAIL');
  if (missing.length){
    console.error('Missing env vars:', missing.join(', '));
    process.exit(1);
  }

}

async function main(): Promise<void> {

  assertEnv();

  const res = await axios.post<BuildResponse>(`${apiUrl}/api/v1/build`,
    { email, vectorsUrl, dimensions },
    { headers: { 'X-TensorTensor-API-Key': apiKey, 'Content-Type': 'application/json' } },
  );

  const { jobId, status } = res.data.data;
  console.log('jobId :', jobId);
  console.log('status:', status);
  console.log('');
  console.log('Save the jobId. You will receive an email when the build completes,');
  console.log('or run 03-end-to-end.ts for an automated submit-and-wait flow.');

}

main().catch((err: any) => {
  console.error('error:', err.response?.data || err.message);
  process.exit(1);
});
